src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -637,28 +637,26 @@ static void window_track_place_draw_mini_preview_track(
 }
 
 static void window_track_place_draw_mini_preview_maze(
-    TrackDesign* td6, int32_t pass, LocationXY16 origin, LocationXY16* min, LocationXY16* max)
+    TrackDesign* td6, int32_t pass, CoordsXY origin, CoordsXY& min, CoordsXY& max)
 {
     uint8_t rotation = (_currentTrackPieceDirection + get_current_rotation()) & 3;
     for (const auto& mazeElement : td6->maze_elements)
     {
-        int16_t x = mazeElement.x * 32;
-        int16_t y = mazeElement.y * 32;
-        rotate_map_coordinates(&x, &y, rotation);
+        CoordsXY mazeCoords{ mazeElement.x * 32, mazeElement.y * 32 };
+        auto rotatedMazeCoords = mazeCoords.Rotate(rotation);
```

Comment:
May as well do the addition as part of this as well to be consistent with the others.